aten/src/ATen/native/native_functions.yaml

```diff
@@ -5844,6 +5844,16 @@
 - func: scatter_add.dimname(Tensor self, Dimname dim, Tensor index, Tensor src) -> Tensor
   variants: function, method
 
+- func: scatter_reduce2(Tensor self, int dim, Tensor index, str reduce, *, int? output_size=None) -> Tensor
```

Comment:
Try naming this "scatter_reduce.2". Our codegen doesn't seem to respect the difference between "." and "_" as it currently stands. Once this lands we should then create an issue to track this.